packages/components/tabs/tests/tabs.test.tsx

```diff
@@ -169,16 +180,15 @@ describe("<Tabs />", () => {
       </Tabs>,
     )
 
-    const tab1 = screen.getByText(/Tab 1/)
-    const tab2 = screen.getByText(/Tab 2/)
-    await act(async () => {
-      fireEvent.keyDown(tab1, { key: "ArrowRight" })
-    })
-    expect(document.activeElement).toStrictEqual(tab2)
+    const tab2 = await screen.findByRole("tab", { name: /Tab 2/i })
+
+    await user.tab()
```

Comment:
Focus on `tab1` is not guaranteed.
The same applies to other. You should not focus with `tab`.

Reply:
@108yen 

It's unfortunate, but even if you replace `await user.tab()` with `fireEvent.focus(tab1)`, the focus does not move to `tab1`, and the test fails.

```tsx
  test("Move to the previous tab with the left arrow key", async () => {
    const { user } = render(
      <Tabs>
        <TabList>
          <Tab>Tab 1</Tab>
          <Tab>Tab 2</Tab>
          <Tab>Tab 3</Tab>
        </TabList>
      </Tabs>,
    )

    const tab1 = await screen.findByRole("tab", { name: /Tab 1/i })
    const tab3 = await screen.findByRole("tab", { name: /Tab 3/i })

    fireEvent.focus(tab1)
    await waitFor(() => {
      expect(tab1).toHaveFocus()
    })

    await user.keyboard("{arrowleft}")
    expect(tab3).toHaveFocus()
  })
```


<details>

<summary>Error log:</summary>

```bash
 FAIL  packages/components/tabs/tests/tabs.test.tsx > <Tabs /> > Move to the previous tab with the left arrow key
Error: expect(element).toHaveFocus()

Expected element with focus:
  <button
  aria-selected="true"
  class="ui-tabs__tab css-76cmc"
  data-index="0"
  role="tab"
  tabindex="0"
  type="button"
>
  Tab 1
</button>
Received element with focus:
  <body
  class="ui-light"
>
  <div>
    <div
      align="start"
      class="ui-tabs css-4rmlxi"
    >
      <div
        class="ui-tabs__list css-w0z0h0"
        role="tablist"
      >
        <button
          aria-selected="true"
          class="ui-tabs__tab css-76cmc"
          data-index="0"
          role="tab"
          tabindex="0"
          type="button"
        >
          Tab 1
        </button>
        <button
          class="ui-tabs__tab css-76cmc"
          data-index="1"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab 2
        </button>
        <button
          class="ui-tabs__tab css-76cmc"
          data-index="2"
          role="tab"
          tabindex="-1"
          type="button"
        >
          Tab 3
        </button>
      </div>
      <div
        class="ui-tabs__panels css-8atqhb"
      />
    </div>
    <span
      hidden=""
      id="__ui_dev"
    />
  </div>
  <div
    class="ui-portal"
  >
    <ul
      class="ui-notice__list ui-notice__list--top css-1nx8pei"
    />
    <ul
      class="ui-notice__list ui-notice__list--top-left css-630uh8"
    />
    <ul
      class="ui-notice__list ui-notice__list--top-right css-8fufzu"
    />
    <ul
      class="ui-notice__list ui-notice__list--bottom css-oote3h"
    />
    <ul
      class="ui-notice__list ui-notice__list--bottom-left css-y29g6d"
    />
    <ul
      class="ui-notice__list ui-notice__list--bottom-right css-1qrpoux"
    />
  </div>
</body>

Ignored nodes: comments, script, style
<html
  data-mode="light"
  data-theme="base"
  style="color-scheme: light;"
>
  <head />
  <body
    class="ui-light"
  >
    <div>
      <div
        align="start"
        class="ui-tabs css-4rmlxi"
      >
        <div
          class="ui-tabs__list css-w0z0h0"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="ui-tabs__tab css-76cmc"
            data-index="0"
            role="tab"
            tabindex="0"
            type="button"
          >
            Tab 1
          </button>
          <button
            class="ui-tabs__tab css-76cmc"
            data-index="1"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Tab 2
          </button>
          <button
            class="ui-tabs__tab css-76cmc"
            data-index="2"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Tab 3
          </button>
        </div>
        <div
          class="ui-tabs__panels css-8atqhb"
        />
      </div>
      <span
        hidden=""
        id="__ui_dev"
      />
    </div>
    <div
      class="ui-portal"
    >
      <ul
        class="ui-notice__list ui-notice__list--top css-1nx8pei"
      />
      <ul
        class="ui-notice__list ui-notice__list--top-left css-630uh8"
      />
      <ul
        class="ui-notice__list ui-notice__list--top-right css-8fufzu"
      />
      <ul
        class="ui-notice__list ui-notice__list--bottom css-oote3h"
      />
      <ul
        class="ui-notice__list ui-notice__list--bottom-left css-y29g6d"
      />
      <ul
        class="ui-notice__list ui-notice__list--bottom-right css-1qrpoux"
      />
    </div>
  </body>
</html>...
 ❯ packages/components/tabs/tests/tabs.test.tsx:172:20
    170|     fireEvent.focus(tab1)
    171|     await waitFor(() => {
    172|       expect(tab1).toHaveFocus()
       |                    ^
    173|     })
    174|
 ❯ runWithExpensiveErrorDiagnosticsDisabled node_modules/.pnpm/@testing-library+dom@10.1.0/node_modules/@testing-library/dom/dist/config.js:47:12
 ❯ checkCallback node_modules/.pnpm/@testing-library+dom@10.1.0/node_modules/@testing-library/dom/dist/wait-for.js:124:77
 ❯ Timeout.checkRealTimersCallback node_modules/.pnpm/@testing-library+dom@10.1.0/node_modules/@testing-library/dom/dist/wait-for.js:118:16
```
</details>

Reply:
This test passes.

```tsx
  test("Move to the previous tab with the left arrow key", async () => {
    const { user } = render(
      <Tabs>
        <TabList>
          <Tab>Tab 1</Tab>
          <Tab>Tab 2</Tab>
          <Tab>Tab 3</Tab>
        </TabList>
      </Tabs>,
    )

    const tab1 = await screen.findByRole("tab", { name: /Tab 1/i })
    const tab3 = await screen.findByRole("tab", { name: /Tab 3/i })

    await user.tab()
    expect(tab1).toHaveFocus()

    await user.keyboard("{arrowleft}")
    expect(tab3).toHaveFocus()
  })
```

Reply:
@koralle
Can't you focus with a click?

Reply:
@108yen 

The test does pass, though. However, it feels a bit off that we have to trigger a `click` event first...

Reply:
@koralle
I agree. But I think `click` is better than `tab`.

Reply:
@108yen 

Well, since we're not asserting the tab panel contents this time, using `click` should be fine. I'll go ahead and make the correction.